src/cljam/util/sam_util.clj

```diff
@@ -161,170 +161,46 @@
-                   [\k \K]    (:k  compressed-bases-high)
-                   [\d \D]    (:d  compressed-bases-high)
-                   [\b \B]    (:b  compressed-bases-high)}))
-
-(defn char->compressed-base-high
-  "Convert from a char to BAM nybble representation of a base in high-order nybble."
-  [base]
-  (or
-   (_char->compressed-base-high base)
-   (throw (IllegalArgumentException. (str "Bad type: " base)))))
-
-(def ^:private _compressed-base->char-low
-  {(:eq compressed-bases-low) \=
-   (:a  compressed-bases-low) \A
-   (:c  compressed-bases-low) \C
-   (:g  compressed-bases-low) \G
-   (:t  compressed-bases-low) \T
-   (:n  compressed-bases-low) \N
-   (:m  compressed-bases-low) \M
-   (:r  compressed-bases-low) \R
-   (:s  compressed-bases-low) \S
-   (:v  compressed-bases-low) \V
-   (:w  compressed-bases-low) \W
-   (:y  compressed-bases-low) \Y
-   (:h  compressed-bases-low) \H
-   (:k  compressed-bases-low) \K
-   (:d  compressed-bases-low) \D
-   (:b  compressed-bases-low) \B})
-
-(defn compressed-base->char-low
-  "Convert from BAM nybble representation of a base in low-order nybble to a char."
-  [base]
-  (or
-   (_compressed-base->char-low (ubyte (bit-and base 0xf)))
-   (throw (IllegalArgumentException. (str "Bad type: " base)))))
-
-(def ^:private _compressed-base->char-high
-  {(:eq compressed-bases-high) \=
-   (:a  compressed-bases-high) \A
-   (:c  compressed-bases-high) \C
-   (:g  compressed-bases-high) \G
-   (:t  compressed-bases-high) \T
-   (:n  compressed-bases-high) \N
-   (:m  compressed-bases-high) \M
-   (:r  compressed-bases-high) \R
-   (:s  compressed-bases-high) \S
-   (:v  compressed-bases-high) \V
-   (:w  compressed-bases-high) \W
-   (:y  compressed-bases-high) \Y
-   (:h  compressed-bases-high) \H
-   (:k  compressed-bases-high) \K
-   (:d  compressed-bases-high) \D
-   (:b  compressed-bases-high) \B})
-
-(defn compressed-base->char-high
-  "Convert from BAM nybble representation of a base in high-order nybble to a char."
-  [base]
-  (or
-   (_compressed-base->char-high (ubyte (bit-and base 0xf0)))
-   (throw (IllegalArgumentException. (str "Bad type: " base)))))
-
-(defn bytes->compressed-bases [^bytes read-bases]
-  (let [rlen (alength read-bases)
-        result-len (quot (inc rlen) 2)
-        result (byte-array result-len)]
-    (loop [i 0]
-      (if (<= result-len i)
-        result
-        (let [h-idx (* 2 i)
-              h (char->compressed-base-high (char (aget read-bases h-idx)))
-              l-idx (inc h-idx)
-              l (if (<= rlen l-idx)
-                  0
-                  (char->compressed-base-low (char (aget read-bases l-idx))))]
-          (aset-byte result i (byte (bit-or h l)))
-          (recur (inc i)))))))
-
-(def ^:const ^:private seq-byte-table
-  (let [nibble-table "=ACMGRSVTWYHKDBN"]
-    (->> (for [i nibble-table j nibble-table] [i j])
-         (apply concat)
-         cstr/join)))
+(defonce ^:private ^:const base->nibble "=ACMGRSVTWYHKDBN")
```

Comment:
Thank you for the comment.
Fixed the names for static tables with comments to be more descriptive.

As sizes of these static tables are not so trivial (<17KB), I used `defonce` to mark that they shouldn't be reallocated / recalculated.
There're no advantages in runtime in the usual case, as you mentioned. I fixed them to `def`.